upgrade/steps.go

```diff
@@ -1021,7 +1021,9 @@ var planPluginSDKUpgrade = stepv2.Func12E("Planning Plugin SDK Upgrade", func(
 	currentBranch, ok := refs.labelOf(currentRef)
 	if !ok {
 		// use latest versioned branch
```

Comment:
It strikes me as very approximate that this code reaches out to terraform-plugin-sdk Git metadata. Why does it need to do that? Is it trying to install the forked terraform-plugin-sdk ahead of the bridge for some reason? Is it possible to instead propagate the replace version from pulumi-terraform-bridge so that the forked terraform-plugin-sdk tracks the bridge it was tested with?

Reply:
>Is it trying to install the forked terraform-plugin-sdk ahead of the bridge for some reason
This is exactly what it does but that seems to be intended IIUC.

I'm happy to look into that but I'd rather ship this first to unblock bridge upgrades.

Reply:
Yeah that's fair we can backlog this. I'm not too comfortable reviewing this codebase but was hoping for some test to match the code changes I thought we're getting more serious about testing this as it is a little hairy? much appreciated.

Reply:
> It strikes me as very approximate that this code reaches out to terraform-plugin-sdk Git metadata. Why does it need to do that? Is it trying to install the forked terraform-plugin-sdk ahead of the bridge for some reason? Is it possible to instead propagate the replace version from pulumi-terraform-bridge so that the forked terraform-plugin-sdk tracks the bridge it was tested with?

It does this because it is updating the `replace`d directive. It can't get picked up by `go mod tidy` because `replace`s are not transitive.

Reply:
We could do some regex magic to get the right version from tf-bridge

Reply:
https://github.com/pulumi/upgrade-provider/issues/245